packages/datadog_session_replay/ios/datadog_session_replay/Sources/Swift/Resource/ResourceRequestBuilder.swift

```diff
@@ -0,0 +1,93 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-Present Datadog, Inc.
+
+import Foundation
+import DatadogInternal
+
+internal struct ResourceRequestBuilder: FeatureRequestBuilder {
+    /// Custom URL for uploading data to.
+    let customUploadURL: URL?
+    /// Sends telemetry through sdk core.
+    let telemetry: Telemetry
+    /// Builds multipart form for request's body.
+    let multipartBuilder: MultipartFormDataBuilder
+
+    init(
+        customUploadURL: URL?,
```

Comment:
nit: I guess it could default to `nil`?
```suggestion
        customUploadURL: URL? = nil,
```

Reply:
It could, but since this is internal I'm okay with it not having a default.